### ISSUE TITLE
Linux

### ISSUE THREAD

**Issue by MyNewBie** [Splamy/TS3AudioBot#98]
hey , is it work on Linux ? if yes , is there any tutorial or video youtube for how to setup this bot on Linux for ts3 ? I really need a perfect musicbot caz sinusbot not friendly with ram and cpu.

**Comment by MrBn100ful** [comment]
Hello, Look the readme.

Dependencies

Any C# Compiler (Visual Studio or mono 5.0.0+ and msbuild)
(Linux only) A C Compiler for Opus
Compilation

Before we start: If you know what you are doing you can alternatively compile each depenency referenced here from source/git by yourself, but I won't add a tutorial for that.

Download the git repository with git clone https://github.com/Splamy/TS3AudioBot.git.

Linux

See if you have NuGet by just executing nuget. If not, get NuGet.exe with wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
Go into the directory of the repository with cd TS3AudioBot
Execute nuget restore or mono ../Nuget.exe restore to download all dependencies
Execute msbuild /p:Configuration=Release /p:Platform=x86 TS3AudioBot.sln to build the C# AudioBot
Make the Opus script runnable with chmod u+x InstallOpus.sh and run it with ./InstallOpus.sh
Get the ffmpeg 32bit or 64bit binary.
Extract the ffmpeg archive with tar -vxf ffmpeg-git-XXbit-static.tar.xz
Get the ffmpeg binary from ffmpeg-git-*DATE*-64bit-static\ffmpeg and copy it to TS3AudioBot/bin/Release/

**Comment by MyNewBie** [comment]
thanks , I did it ok , now bot connected on my ts3 so where I must put my songs ? I mean which folder ? and how I can add song by url ? and final question is there any web panel for this bot ?

**Comment by MrBn100ful** [comment]
Just read the wiki ! 
https://github.com/Splamy/TS3AudioBot/wiki

RTFM.

**Comment by MyNewBie** [comment]
I pasted a music.mp3 behind the /TS3AudioBot/bin/Release/TS3AudioBot.exe and I tried to send !play ./music.mp3 on private message , the console reply "AF ar start: MediaLink ID:./music.mp3" but nothing happened in TeamSpeak , can you explain more ?

**Comment by MrBn100ful** [comment]
the command is" !play music.mp3 "

**Comment by MyNewBie** [comment]
work perfectly , is there possible to add radio ?

**Comment by Splamy** [comment]
same, !play

**Comment by davlast** [comment]
some radio stations do work, just need the right url
!play http://pulseedm.cdnstream1.com:8124/1373_128
!play http://20133.live.streamtheworld.com:80/SKYRADIO_SC

**Comment by MyNewBie** [comment]
I changed my vps , I did every thing again but i have new problem when i want run it : 
https://pastebin.com/WU6RkRMM
please check the log.

**Comment by Splamy** [comment]
Can you tell me what your `QueryConnection::host` field says
and which version do you have

**Comment by MyNewBie** [comment]
87.236.209.72 this is my ts ip server

**Comment by Splamy** [comment]
1. can you give me the the entire line
1. and which version did you get

**Comment by MyNewBie** [comment]
i get the last source form your repo .

my cfg : https://pastebin.com/qgwVpf9B

**Comment by Splamy** [comment]
set `MainBot::BotGroupId` back to `0` the bot will set it with the `!bot setup` later.
Can you try the latest develop branch?
You can get it with `git clone -b develop https://github.com/Splamy/TS3AudioBot`
Or checkout with `git checkout develop` if you already have the git repo

**Comment by MyNewBie** [comment]
ok im trying

**Comment by MyNewBie** [comment]
oh yes :D perfectly run ok without any problem xD thanks for answer speedly

**Comment by MyNewBie** [comment]
ok , can i come much more than one bot ? i need 5 bot on my server and how i can create a playlist or stream radio ?
and whisper spam is Annoying how i can disable it ?

**Comment by Splamy** [comment]
at the ts3ab program can only start one bot. You'd have to start the exe multiple times, though this might be a bit instable.
You can create playlists with the `!playlist` command. See `!help playlist` for more details.
You can play any (radio-)stream with `!play <link>`
You can disable whisper and use normal voice mode by setting `AudioFramework::AudioMode=whisper` to `AudioFramework::AudioMode=voice`

**Comment by MyNewBie** [comment]
thanks you can close this issue now .

**Comment by MyNewBie** [comment]
sorry I reopen again . I have some question  : 
is there any forum ? so I will post my question in there .
1- i create a playlist so i want send name of music when a track get start play . is there any ability ? if yes how i can enable it ?
2- is there any config for join the bot to specific channel i want ( after start the bot) ?
3- i read the wiki for "rights.toml" bot i don't understand it good , so is there any simple of "rights.toml" so i can write my own file.(if you have one please share)
4- is there any function for welcome sound ? play a music , when someone join the server get music volume down then play welcome sound then get volume music up again .

**Comment by Splamy** [comment]
We dont have a forum, feel free to ask all questions here
1. Not yet, I'll see if I can add it in near future.
1. Also not yet. This is a feature which already was requested in #85 
1. The bot will create a default toml file with a few comments, you can look through there and play around with that.
When you want to debug your rights, there are two convenient commands:
    - `!rights reload` to reaload the file at runtime
    - `!rights can <list of rights>` Which filters from a list of rights all rights you can use. For example `!rights can cmd.help cmd.play` will answer you with `cmd.help, cmd.play` if you have both, or `cmd.help` if you only have this one, etc.
1. Also not possible. This seems to be a kind of nieche request, so I will put it in the backlog

**Comment by MyNewBie** [comment]
hey again , I created folder beside TS3AudioBot.exe I called it Playlist so I uploaded some music.mp3 file in there , so how I can play all of them ? I used `!add /Playlist/music.mp3 ` and not work I tried `!play /Playlist/music.mp3` not work too , I really don't want add my music list every each restart vps  just add them once time then play folder . is it possible ?

**Comment by Flakebi** [comment]
Someone from another issue made a [list of playlist commands](https://github.com/Splamy/TS3AudioBot/issues/106#issuecomment-324133039).
With these commands and `!list add <link>` (write `!help list` for the full command list and `!help list add` for a more detailed explanation) you can create your own playlists.

Playing local files (and adding them to playlists) is also possible, you just have to insert the path (like `!play Playlists/music.mp3` or `!list add Playlists/music.mp3`).

**Comment by MyNewBie** [comment]
ok so how I can add music name like this  `my music name.mp3` music name with some white space .

**Comment by Flakebi** [comment]
Just put it into quotes: `!play "music folder/my music name.mp3"`

**Comment by MyNewBie** [comment]
ok work , so I tried `!list list` replayed with my songs name , but `!list play` didn't work my play list

**Comment by Flakebi** [comment]
`!list list` shows all available playlists.
You can choose one with `!list load <name>` (e.g. `!list load "my new playlist"`)
and then play it using `!list play`.

**Comment by MyNewBie** [comment]
ok I tried `!list load "/home/TS3AudioBot/TS3ABotUnitTests/bin/Release/Playlists/Musiclists"` I got this `Loaded: "/home/TS3AudioBot/TS3ABotUnitTests/bin/Release/Playlists/Musiclists" with 0 songs`

**Comment by Flakebi** [comment]
Just try `!list load Musiclists` or whatever is listed in `!list list`.

**Comment by MyNewBie** [comment]
ok with `!list load Musiclists` I got this `Loaded: "Musiclists" with 0 songs` but with !list list I got `my first song.mp3,my second song.mp3,Musiclists`

**Comment by MyNewBie** [comment]
i added 250 sound track to playlists folder and i added every each track with `!list add <music_path/music.mp3>` so after vps reset i need to add all of them again ? or i can save playlist and add new track later ?

**Comment by Flakebi** [comment]
For the warning: It does not look familiar to me, we have to wait until @Splamy is back. Some lines above and below the warning would be nice though.

I found out that you can use `!list get <folder>` to create a new playlist with all songs in a specific folder.
Yes, `!list save` (or `!list save <name>`) save the list to disk, so it can be loaded again afterwards.

I assembled a list of playlist commands and explanations [in the wiki](https://github.com/Splamy/TS3AudioBot/wiki/Playlists) (although it's not yet complete).

**Comment by MyNewBie** [comment]
ok thanks @Flakebi is it possible to set bot as channel commander ( orange light ) ? i used `!random on` for my playlist and it work ok but about `!repeat on` for play again my playlist after finish all of songs im not sure (i didnt test).

**Comment by Flakebi** [comment]
It seems like `!repeat` does only repeat a single song. `!loop` should work for entire playlists.

The bot cannot set itself as a channel commander at the moment. I don't know how much work it would be to implement it.

**Comment by MyNewBie** [comment]
about radio `!play http://88.198.69.145/` not work `Error: Could not load (MediaNoWebResponse)` but with foobar in my pc it work ok . why ?

**Comment by davlast** [comment]
try this, 
`!play http://88.198.69.145:7355/live?1503578655875`

**Comment by Splamy** [comment]
This is just the website, seems like foobar crawls the site for valid stream links/stream container.
You'd have to manually search the site for the stream link which looks. For this site for example if found this `http://88.198.69.145:7355/live`

**Comment by MyNewBie** [comment]
sorry the correct http is `http://208.85.241.142` how I can find stream link for this one ?

**Comment by Splamy** [comment]
Works right out of the box? `!play http://208.85.241.142`

**Comment by MyNewBie** [comment]
but with browser it open windows media player and work ok in my pc

**Comment by MyNewBie** [comment]
ok , thanks

**Comment by MyNewBie** [comment]
have any idea how can set client to channel commander ? I just want orange bubble not blue, maybe change something from client ?

**Comment by Splamy** [comment]
I'll look into it, when I find the correct command it shouldn't be too difficilt

**Comment by MyNewBie** [comment]
would be cool .  thanks

**Comment by Splamy** [comment]
Feature is up on the latest dev branch, you can enable it with `!bot commander on`

**Comment by MyNewBie** [comment]
woW , great @Splamy thanks .

**Comment by Splamy** [comment]
There's no option in the config for that, and I dont think that would be needed.
On the other hand I've planned to add a few options to execute ABS (=AudioBotScript) for various events. You can for example already execute a custom script when a new song starts, but I wannt to add other events like connecting(/startup), song stop.
This way the config would not get bloatet by hundreds of small (mostly irrelevant) settings but can still be customized for everyones needs.

**Comment by MyNewBie** [comment]
default channel config needed I guess , because vps reset. idk

**Comment by Splamy** [comment]
There is a config to join a default channel on start

**Comment by MyNewBie** [comment]
and whats that config name ?

**Comment by davlast** [comment]
you can configure it in ``configTS3AudioBot.cfg``

``
# Default Channel when connecting
# Use a channel path or '/<id>', examples: 'Home/Lobby', '/5', 'Home/Afk \/ Not$
QueryConnection::DefaultChannel=
``
[#85]https://github.com/Splamy/TS3AudioBot/issues/85

**Comment by Splamy** [comment]
All new config values are created automatically when you start the bot once. This has the advantage that an explaining comment will be generated too.
The new setting is called: `QueryConnection::DefaultChannel`
Comment:
```
# Default Channel when connectiong
# Use a channel path or '/<id>', examples: 'Home/Lobby', '/5', 'Home/Afk \/ Not Here'
```

**Comment by Splamy** [comment]
Since this issue is older than 2 weeks with no open questions I will close it.
Feel free to reopen or open a new issue if you have any further questions or problems.
